### ISSUE TITLE
v0.10.0 finds errors in react-native example

### ISSUE THREAD

**Issue by davidpfahler** [facebook/flow#423]
I don't know if this is a bug or expected, but I couldn't find anything in the documentation. This is related to [this issue](https://github.com/facebook/react-native/issues/938) on react-native.

To reproduce this issue:
* create a new react-native project using `react-native init testflow`
* `cd testflow`
* `flow --json`

Output:
```
flow is outdated, going to launch a new one.
The flow server will be ready in a moment
Flow server launched for /Users/djpfahler/localhost/testflow
Spawned flow server (child pid=12001)
Logs will go to /var/folders/mh/lh59l19x601d96hkhgxk406h0000gn/T/flow/zSUserszSdjpfahlerzSlocalhostzStestflow.log
The flow server will be ready in a moment.
{"passed":false,"errors":[{"message":[{"descr":"undefined\nThis type is incompatible with","code":0,"path":"/Users/djpfahler/localhost/testflow/node_modules/react-native/Libraries/Components/TabBarIOS/TabBarItemIOS.ios.js","line":70,"endline":70,"start":15,"end":34},{"descr":"boolean","code":0,"path":"/Users/djpfahler/localhost/testflow/node_modules/react-native/Libraries/Components/TabBarIOS/TabBarItemIOS.ios.js","line":91,"endline":91,"start":62,"end":68}]},{"message":[{"descr":"undefined\nThis type is incompatible with","code":0,"path":"/Users/djpfahler/localhost/testflow/node_modules/react-native/Libraries/ReactIOS/ReactIOSMount.js","line":197,"endline":197,"start":7,"end":13},{"descr":"boolean","code":0,"path":"/Users/djpfahler/localhost/testflow/node_modules/react-native/Libraries/ReactIOS/ReactIOSMount.js","line":194,"endline":194,"start":59,"end":62}]}],"version":" Apr 24 2015 18:33:55"}
```

This was not the case with `flow@0.9.2`, as can be reproduced by downloading the 0.9.2 executable and then run `~/Downloads/flow-0.9.2/flow --json` in the `testflow` directory which will output:

```
flow is outdated, going to launch a new one.
The flow server will be ready in a moment
Flow server launched for /Users/djpfahler/localhost/testflow
Spawned flow server (child pid=12240)
Logs will go to /var/folders/mh/lh59l19x601d96hkhgxk406h0000gn/T/flow/zSUserszSdjpfahlerzSlocalhostzStestflow.log
The flow server will be ready in a moment.
{"passed":true,"errors":[],"version":" Apr 17 2015 16:01:38"}an take some time. Retrying... -
```

Please make sure you intentionally "broke" this. I would also appreciate if you could comment on https://github.com/facebook/react-native/issues/938. 

Thank you a lot. :)

**Comment by gabelevi** [comment]
Hey @davidpfahler, thanks a lot of the report. I'll make sure to comment on https://github.com/facebook/react-native/issues/938 as well.


Almost every release of Flow includes some breaking changes, as we make Flow smarter and it notices things it missed before. It seems that the version of react-native on GitHub still expects to work with Flow v0.9.2. Flow v0.10.0 complains about a few more things, and react-native needs a few small updates in order to work without any Flow errors. We've made those changes, but it may take a few days for those changes to make it to GitHub. In the meantime, Flow v0.9.2 should still work.

What we clearly have missed here is a good way of tying Flow's versions to those expected by react-native. If Flow was just a npm package, we would specify the specific Flow version expected in the package.json and be done with it. We're talking about this internally to figure out the best thing to do moving forward. Sorry for the trouble and thanks for the report!

**Comment by gabelevi** [comment]
[react-native 0.4.1](https://www.npmjs.com/package/react-native) has been pushed to npm, and it includes the fixes for Flow v0.10.0. This week we'll invest in making sure this is much smoother for future deploys. Thanks again for reporting!
